src/utils.jl

```diff
@@ -10,6 +10,30 @@ macro check_args(K, param, cond, desc=string(cond))
 end
 
 
+"""
+    ColVecs{T, TX<:AbstractMatrix}
+
+A lightweight box for an `AbstractMatrix` to make it behave like a vector of vectors.
+"""
+struct ColVecs{T, TX<:AbstractMatrix{T}} <: AbstractVector{Vector{T}}
+    X::TX
+    ColVecs(X::TX) where {T, TX<:AbstractMatrix{T}} = new{T, TX}(X)
+end
+
+Base.:(==)(D1::ColVecs, D2::ColVecs) = D1.X == D2.X
+Base.size(D::ColVecs) = (size(D.X, 2),)
+Base.length(D::ColVecs) = size(D.X, 2)
+Base.getindex(D::ColVecs, n::Int) = D.X[:, n]
+Base.getindex(D::ColVecs, n::CartesianIndex{1}) = getindex(D, n[1])
+Base.getindex(D::ColVecs, n) = ColVecs(D.X[:, n])
+Base.view(D::ColVecs, n::Int) = view(D.X, :, n)
+Base.view(D::ColVecs, n) = ColVecs(view(D.X, :, n))
+Base.eltype(D::ColVecs{T}) where T = Vector{T}
```

Comment:
`eltype` should be defined for types not instances (see https://docs.julialang.org/en/v1/base/collections/#Base.eltype):
```suggestion
Base.eltype(::Type{<:ColVecs{T}}) where T = Vector{T}
```